### PR TITLE
Add support for loading modules and using members from them

### PR DESCRIPTION
**Pull request by nex3**
See https://github.com/sass/sass-spec/pull/1349

### CODE REVIEW

**Comment by nex3** [comment]
You only need to review [the commits after the merge](https://github.com/sass/dart-sass/pull/603/files/7a7aea95b345f144c3cdd9f17632b85bc635254c..5a64dec4ec54ef4fc8fdf0ca0c16a83fec81463b); everything else was already reviewed when it landed on `master`.

File: lib/src/parse/stylesheet.dart [hunk 1]
```diff
@@ -71,6 +71,15 @@ abstract class StylesheetParser extends Parser {
   /// Whether the parser is currently within a parenthesized expression.
   var _inParentheses = false;
 
+  /// A map from all variable names that are assigned with `!global` in the
+  /// current stylesheet to the nodes where they're defined.
+  ///
+  /// These are collected at parse time because they affect the variables
+  /// exposed by the module generated for this stylesheet, *even if they aren't
+  /// evaluated*. This allows us to ensure that the kstylesheet always exposes
```
Comment (nshahan) [inline lib/src/parse/stylesheet.dart]: typo: "kstylesheet"

Comment (nex3) [reply lib/src/parse/stylesheet.dart]: Done.

File: lib/src/importer/utils.dart [hunk 2]
```diff
@@ -6,22 +6,61 @@ import 'package:path/path.dart' as p;
...
 import '../io.dart';
 
+/// Whether the Sass interpreter is currently evaluating a `@use` rule.
+///
+/// The `@use` rule has slightly different path-resolution behavior than
+/// `@import`: `@use` prioritizes a `.css` file with a given name at the same
+/// level as `.sass` and `.scss`, while `@import` prefers `.sass` and `.scss`
+/// over `.css`. It's admittedly hacky to set this globally, but `@import` will
+/// eventually be removed, at which point we can delete this and have one
+/// consistent behavior.
```
[the remark this answers is not in the document]
Comment (nex3) [reply lib/src/importer/utils.dart]: That's a good question. LibSass, which incorrectly supports `.css` files right now, *does* produce a conflict error here, but that doesn't necessarily mean it's the best option. Let's talk this through in person on Monday.

[the remark this answers is not in the document]
Comment (nshahan) [reply lib/src/importer/utils.dart]: It's ok with me if you want to commit these changes and revise how you see fit per [language/68](https://github.com/sass/language/issues/68).

**Pull request merged**